src/InfoMCStructHelper.cc

```diff
@@ -171,22 +171,37 @@ namespace mu2e {
 
   void InfoMCStructHelper::fillAllSimInfos(const KalSeedMC& kseedmc, const PrimaryParticle& primary, std::vector<std::vector<SimInfo>>& all_siminfos, int n_generations, int n_match) {
     std::vector<SimInfo> siminfos;
-
+    
     // interpret -1 as no llimit
     if (n_generations == -1) {
       n_generations = std::numeric_limits<int>::max();
     }
     if (n_match == -1 ) {
       n_match = std::numeric_limits<int>::max();
     }
+    
+    //create vector of art Ptr to particles:
+    std::vector<art::Ptr<SimParticle> > allParts;
+
+    //loop over all sime particles in KalSeedMC
     for(int imatch = 0 ; imatch < std::min(n_match,static_cast<int>(kseedmc.simParticles().size())); ++imatch) {
       auto trkprimaryptr = kseedmc.simParticle(imatch).simParticle(_spcH);
       auto trkprimary = trkprimaryptr->originParticle();
       auto current_sim_particle_ptr = trkprimaryptr;
-      auto current_sim_particle = trkprimary;
-
+      auto current_sim_particle = trkprimary; 
+      
       for (int i_generation = 0; i_generation < n_generations; ++i_generation) {
+        bool isSame = false;
+        for(unsigned int ipart = 0; ipart < allParts.size() ; ipart++){
+          MCRelationship checkrel(current_sim_particle_ptr,allParts.at(ipart));
+          if(checkrel==MCRelationship::same) {
+              isSame = true;
+            }
+        }
+        if(!isSame) allParts.push_back(current_sim_particle_ptr);
```

Comment:
Could you add the curly braces around this (it can still be on one line)? I know it doesn't make a difference but I prefer to be explicit rather than rely on the compiler to add them